src/mapper.rs

```diff
@@ -46,4 +46,39 @@ impl CodeMapper {
     pub fn heap_bytes(&self) -> usize {
         self.table.len() * core::mem::size_of::<u32>()
     }
+
+    #[inline]
+    pub fn io_bytes(&self) -> usize {
+        self.heap_bytes() + core::mem::size_of::<u32>() * 2
```

Comment:
`io_bytes()` should not be calculated using `heap_bytes()`.
Could you calculate this from scratch?